File: c/2302.c
long long countSubarrays(int* n, int l, long long k) {
    long long c = 0, s = 0;
    int p = -1;
    for (int i = 0; i < l; i++) {
        s += n[i];
        while ((s * (i - p)) >= k)
            s -= n[++p];
        c += i - p;
    }
    return c;
}

File: python3/2302.py
class Solution:
    def countSubarrays(self, n: List[int], k: int) -> int:
        c = 0
        s = 0
        p = -1
        for i in range(len(n)):
            s += n[i]
            while (s * (i - p)) >= k:
                p += 1
                s -= n[p]
            c += i - p
        return c
